nextflow.config

```
@@ -296,6 +296,8 @@ profiles {
     }
 
     // Extra test profiles for more complete CI
+    // TO-DO: Indentation!
+    alignment_from_fastq_and_spring { includeConfig 'conf/test/alignment_from_fastq_and_spring.config' }
```

```suggestion
    alignment_from_spring { includeConfig 'conf/test/alignment_from_spring.config' }
```

Why not keep it simple?

---

I wanted to be thorough with the test and add a test-csv with both fastq-files and spring-files :-) The config/profile name ` alignment_from_fastq_and_spring` reflects that.  Anyways, I'm also fine with your suggestion.

---

I see.
so for mapping we can start with FASTQ, SPRING, FASTQ and SPRING, BAM, can we do FASTQ SPRING and BAM?

---

Or maybe the question is SHOULD WE?

---

I haven't tried FASTQ+SPRING+BAM - not that brave 😆

---

not that brave, or not that stupid?

---

Let's just say I braved it 😁
```
nextflow run main.nf -profile test,alignment_from_fastq_and_spring,docker --outdir results --save_mapped --save_output_as_bam
```
with this input-csv:
```
patient,sex,status,sample,lane,fastq_1,fastq_2,spring_1,spring_2,bam,bai
test,XX,0,test,test_L1,https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz,https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz,,,,
test2,XX,0,test2,test2_L1,,,https://raw.githubusercontent.com/nf-core/test-datasets/sarek3/data/genomics/homo_sapiens/illumina/spring/test_1.fastq.gz.spring,https://raw.githubusercontent.com/nf-core/test-datasets/sarek3/data/genomics/homo_sapiens/illumina/spring/test_2.fastq.gz.spring,,
test3,XX,0,test3,test3_L1,,,https://raw.githubusercontent.com/nf-core/test-datasets/sarek3/data/genomics/homo_sapiens/illumina/spring/test_R1_R2.fastq.gz.spring,,,
test_bam,XX,0,test_bam,test_bam_L1,,,,,https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam,https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai
```
and I got:
```
├── mapped
│   ├── test
│   │   ├── test.sorted.bam
│   │   └── test.sorted.bam.bai
│   ├── test2
│   │   ├── test2.sorted.bam
│   │   └── test2.sorted.bam.bai
│   ├── test3
│   │   ├── test3.sorted.bam
│   │   └── test3.sorted.bam.bai
│   └── test_bam
│       ├── test_bam.sorted.bam
│       └── test_bam.sorted.bam.bai
├── markduplicates
│   ├── test
│   │   ├── test.md.bam
│   │   └── test.md.bam.bai
│   ├── test2
│   │   ├── test2.md.bam
│   │   └── test2.md.bam.bai
│   ├── test3
│   │   ├── test3.md.bam
│   │   └── test3.md.bam.bai
│   └── test_bam
│       ├── test_bam.md.bam
│       └── test_bam.md.bam.bai
├── recal_table
│   ├── test
│   │   └── test.recal.table
│   ├── test2
│   │   └── test2.recal.table
│   ├── test3
│   │   └── test3.recal.table
│   └── test_bam
│       └── test_bam.recal.table
└── recalibrated
    ├── test
    │   ├── test.recal.bam
    │   └── test.recal.bam.bai
    ├── test2
    │   ├── test2.recal.bam
    │   └── test2.recal.bam.bai
    ├── test3
    │   ├── test3.recal.bam
    │   └── test3.recal.bam.bai
    └── test_bam
        ├── test_bam.recal.bam
        └── test_bam.recal.bam.bai
 ```

---

@maxulysse : So `alignment_from_everything`? 😁

---

Yeahhhh let's go

---

You also want cram+crai in there?

---

I'm saying that might be too much, let's stop there...

@FriederikeHanssen are we starting to get crazy/stupid/brave or all of the above?

---

Unless I get some protests like very soon, I'll be adding a pytests for `alignment_from_everything` (that is actually everything except cram+crai).

---

> @FriederikeHanssen are we starting to get crazy/stupid/brave or all of the above?

should have some time today to review this in more detail and let you know 😆